### PR TITLE
Breaking

### PR DESCRIPTION
**Pull request by notlee**
- Add primary mixin `oLazyLoad`
- Remove public Sass which has been replaced by `oLazyLoad`.
- Remove the ability to customise CSS class names.
- Add `o-lazy-load--transition` class. Previously a mixin
could be called to make lazy loaded content transition from
the placeholder with opacity. This applied to the `o-lazy-load` class.
If a project dependency were to call that mixin it would effect
all lazy loaded items unintentionally.

See https://github.com/Financial-Times/o-lazy-load/issues/12

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-lazy-load bundle size difference from 1.0.2 to 5638c27f80deb161ae81d6fecf2c0b2bcb750637
No significant bundle size differences found.

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+```scss
+@include oLazyLoad($opts: (
+    'placeholder': true, // e.g. .o-lazy-load-placeholder
+    'placeholder-ratios': ((16, 9), (1, 1)), // e.g. .o-lazy-load-placeholder--16x9
+    'transition': true
+));
+```
+
+`$o-lazy-load-placholder-ratios` is now private and must not be used, instead use the `oLazyLoad` option map above.
```
Comment (JakeChampion) [inline MIGRATION.md]:
```suggestion
`$o-lazy-load-placeholder-ratios` is now private and must not be used, instead use the `oLazyLoad` option map above.
```

File: test/scss/_mixins.test.scss [hunk 2]
```diff
@@ -1,56 +1,84 @@
...
+	@include it('outputs custom placeholder ratios') {
+		@include assert() {
+			@include output() {
+				@include oLazyLoad($opts: (
+					'placeholder': true,
+					'placeholder-ratios': ((32, 9), (4, 3))
+				));
+			};
+
+			@include contains() {
```
Comment (JakeChampion) [inline test/scss/_mixins.test.scss]: Woo!

**Comment by origamiserviceuser** [comment]
o-lazy-load bundle size difference from 1.0.2 to 21b881071a0e4ac6a19d66204f616022eeb61aee
No significant bundle size differences found.

**Pull request merged**